guides/session.rst

```diff
@@ -80,42 +118,78 @@ With ``Behat\Mink\Session`` you can control your browsers cookies and headers:
     // retrieving response headers:
     print_r($session->getResponseHeaders());
 
-    // set cookie:
-    $session->setCookie('cookie name', 'value');
-
-    // get cookie:
-    echo $session->getCookie('cookie name');
-
-    // delete cookie:
-    $session->setCookie('cookie name', null);
-
 .. note::
 
     Headers handling is only supported in headless drivers, because there
     is no way browser controllers can get such information out of the browser.
 
 HTTP Authentication
-~~~~~~~~~~~~~~~~~~~
+-------------------
 
-Also, Mink session has a special method to perform HTTP Basic authentication:
+The Mink session has a special method to perform HTTP Basic authentication:
 
 .. code-block:: php
 
     $session->setBasicAuth($user, $password);
 
+The method can also be used to reset a previous authentication:
+
+.. code-block:: php
+
+    $session->setBasicAuth(false);
+
 .. note::
 
     Automatic HTTP authentication is only supported in headless drivers.
     Because HTTP authentication in browser requires manual user action, that
     can't be done remotely for browser controllers.
 
+Javascript Evaluation
+---------------------
+
+The session allows you to execute or evaluate Javascript.
+
+.. code-block:: php
+
+    // Execute JS
+    $session->evaluateScript('document.body.firstChild.innerHtml = "";');
+
+    // evaluate JS expression:
+    echo $session->evaluateScript(
+        "return 'something from browser';"
+    );
+
+.. note::
+
+    The difference between these methods is that ``Session::evaluateScript``
+    returns the result of the expression. When you don't need to get a return
+    value, using ``Session::executeScript`` is better.
+
+You can also wait until a give JS expression returns a truthy value or the
+timeout is reached:
+
+.. code-block:: php
+
+    // wait for n milliseconds or
+    // till JS expression becomes true:
+    $session->wait(
+        5000,
+        "$('.suggestions-results').children().length"
+    );
+
+.. note::
+
+    The ``Session::wait`` method returns the result of the evaluation. It
```

Comment:
That's not true (at least for Selenium2Driver) because `wait` method always returns `true` or `false` to indicate that:
* `true` - what we're waiting for has happened
* `false` - what we're waiting for hasn't happened (also in case of timeout)